src/internals/tuples/Tuples.ts

```diff
@@ -159,9 +159,16 @@ export namespace Tuples {
   > = Functions.PartialApply<MapFn, [fn, tuple]>;
 
   interface MapFn extends Fn {
-    return: ReduceImpl<this["arg1"], [], MapReducer<Extract<this["arg0"], Fn>>>;
+    return: this["arg1"] extends [...any[]]
+      ? MapFnImpl<this["arg1"], Extract<this["arg0"], Fn>>
+      : never;
   }
 
+  // This needs to be extracted out to show as a tuple type in editor tooling.
+  type MapFnImpl<a extends [...any[]], f extends Fn> = {
+    [K in keyof a]: Call<f, a[K]>;
+  };
+
```

Comment:
Be carreful with naming. don't use single letters if possible. also `[..any[]]` should not be necessary.

Also i measured simple use cases and don't see perf improvements overall.

```suggestion
    return: MapFnImpl<this["arg1"], Extract<this["arg0"], Fn>>;
  }

  type MapFnImpl<tuple extends unknown[], fn extends Fn> = {
    [K in keyof tuple]: Call<fn, tuple[K]>;
  };
```
```

Reply:
I have updated the code. Could've sworn that you needed `[...any[]]` to get it to work properly, but you're right it does work without it. Also, the benefits of this is that we are no longer limited by the recursion limit (10,000) and instead the union size limit which seems to be greater than 50,000.